src/compile/scale.ts

```diff
@@ -68,15 +69,29 @@ export function parseScaleComponent(model: Model): Dict<ScaleComponents> {
 function parseMainScale(model: Model, fieldDef: FieldDef, channel: Channel) {
   const scale = model.scale(channel);
   const sort = model.sort(channel);
-
   let scaleDef: any = {
     name: model.scaleName(channel),
     type: scale.type,
   };
 
-  scaleDef.domain = domain(scale, model, channel);
-  extend(scaleDef, rangeMixins(scale, model, channel));
+  // If channel is either X or Y then union them with X2 & Y2 if they exist 
+  if (channel === X && model.has(X2)) {
```

Comment:
I think the logic here is correct.  That said, I want to cc: @domoritz  as this might have implication for how we merge scale later down the stream.